app/models/veteran_enrollment_system/form1095_b/form1095_b.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+require 'common/models/resource'
+
+module VeteranEnrollmentSystem
+  module Form1095B
+    class Form1095B
+      include Vets::Model
+
+      attribute :first_name, String
+      attribute :middle_name, String
+      attribute :last_name, String
+      attribute :last_4_ssn, String
+      attribute :birth_date, Date
+      attribute :address, String
+      attribute :city, String
+      attribute :state, String
+      attribute :province, String
+      attribute :country, String
+      attribute :zip_code, String
+      attribute :foreign_zip, String
+      attribute :is_corrected, Bool, default: false
+      attribute :coverage_months, Array
+      attribute :tax_year, String
+
+      def txt_file
+        template_path = self.class.txt_template_path(tax_year)
+        unless File.exist?(template_path)
+          Rails.logger.error "1095-B template for year #{tax_year} does not exist."
+          raise Common::Exceptions::UnprocessableEntity.new(
+            detail: "1095-B for tax year #{tax_year} not supported", source: self.class.name
+          )
+        end
+
+        template_file = File.open(template_path, 'r')
+        template_data = attributes.merge(txt_form_data)
+        prepared_text = template_file.read % template_data.symbolize_keys
+        template_file.close
+
+        prepared_text
+      end
+
+      def pdf_file
+        template_path = self.class.pdf_template_path(tax_year)
+        unless File.exist?(template_path)
+          Rails.logger.error "1095-B template for year #{tax_year} does not exist."
+          raise Common::Exceptions::UnprocessableEntity.new(
+            detail: "1095-B for tax year #{tax_year} not supported", source: self.class.name
+          )
+        end
+
+        pdftk = PdfForms.new(Settings.binaries.pdftk)
+        tmp_file = Tempfile.new("1095B-#{SecureRandom.hex}.pdf")
+        generate_pdf(pdftk, tmp_file, template_path)
+      end
+
+      class << self
+        # there is some overlap in the data provided by coveredIndividual and responsibleIndividual.
+        # in the VA enrollment system, they are always the same.
+        def parse(form_data)
+          prepared_data = {
+            first_name: form_data['data']['coveredIndividual']['name']['firstName'],
+            middle_name: form_data['data']['coveredIndividual']['name']['middleName'],
+            last_name: form_data['data']['coveredIndividual']['name']['lastName'],
+            last_4_ssn: form_data['data']['coveredIndividual']['ssn'],
+            birth_date: form_data['data']['coveredIndividual']['dateOfBirth'],
+            address: form_data['data']['responsibleIndividual']['address']['street1'],
+            city: form_data['data']['responsibleIndividual']['address']['city'],
+            state: form_data['data']['responsibleIndividual']['address']['stateOrProvince'],
+            province: form_data['data']['responsibleIndividual']['address']['stateOrProvince'],
```

Comment:
@kpethtel Would it be possible to add logic to determine whether address is US (populate state) or foreign (populate province)? Thanks!

Reply:
This logic is copied from the existing logic in the original 1095b model. There's probably room for improvement, but the current logic seems to work so I'm reluctant to touch it.